internal/prober/http/http.go

```diff
@@ -252,6 +260,25 @@ func convertOAuth2Config(ctx context.Context, cfg *sm.OAuth2Config, logger zerol
 	return r, nil
 }
 
+// Overrides any user-provided headers with our own augmented values
+// for 'reserved' headers.
+func augmentHttpHeaders(check *sm.Check, additionalHeaders []string) {
+	headers := []string{}
+	for _, header := range check.Settings.Http.Headers {
+		name, _ := strToHeaderNameValue(header)
+
+		if strings.ToLower(name) == checkProbeIdHeader {
```

Comment:
Following my previous comment, it feels odd to check for the specific `checkProbeIdHeader` here. Would it make sense to accept as input an `http.Header` field, which identifies the "reserved headers", and then, for any header set in the check that is already present in the reserved ones, we forbid it?

Reply:
Absolutely!